instrumentation/grpc-1.6/library/src/main/java/io/opentelemetry/instrumentation/grpc/v1_6/GrpcTracingBuilder.java

```diff
@@ -49,9 +52,28 @@ public GrpcTracingBuilder addAttributeExtractor(
     return this;
   }
 
+  /**
+   * Sets custom client {@link SpanNameExtractor}
+   */
+  public GrpcTracingBuilder setClientSpanNameExtractor(
+      SpanNameExtractor<? super GrpcRequest> clientSpanNameExtractor) {
```

Comment:
Instead of just passing the `SpanNameExtractor`, WDYT about passing a `Function<SpanNameExtractor<GrpcRequest>, ? extends SpanNameExtractor<? super GrpcRequest>>`? (similar to how [the Armeria instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/ArmeriaTracingBuilder.java#L55) allows modifying the `SpanStatusExtractor`)
With a transformer function you can still make use of the original span name extractors - or reject them altogether and return your own, if you wish to do so.

Reply:
Sounds interesting
Chaining status code transformations makes more sense imo than extracting span name (which is a string, doubt you'd like to have pattern matching on it)

Or what's the use case you have in mind? One could be if length is more that X use shorter version of method name; since default impl is pretty straight forward I'd stick to the current version, + it gives api consistency with other methods

Reply:
> Or what's the use case you have in mind? One could be if length is more that X use shorter version of method name;

That's one option. I was thinking of a case where you might decide to fall back on the default naming scheme based on the request, e.g. `(request) -> shouldRename(request) ? customName(request) : originalSpanNameExtractor.extract(request)`

Reply:
@trask what do you think? current way as api consistency vs suggested flexibility? 

I'm ok with both

Reply:
let's go with @mateuszrzeszutek's proposal, thx!